src/dotnet-install.ps1

```diff
@@ -289,16 +289,34 @@ function GetHTTPResponse([Uri] $Uri)
             $Task = $HttpClient.GetAsync("${Uri}${FeedCredential}").ConfigureAwait("false");
             $Response = $Task.GetAwaiter().GetResult();
             if (($Response -eq $null) -or (-not ($Response.IsSuccessStatusCode))) {
-                 # The feed credential is potentially sensitive info. Do not log FeedCredential to console output.
-                $ErrorMsg = "Failed to download $Uri."
+                # The feed credential is potentially sensitive info. Do not log FeedCredential to console output.
+                $ErrorMsg = ""
                 if ($Response -ne $null) {
-                    $ErrorMsg += "  $Response"
+                    $ErrorMsg += "$Response"
                 }
-
                 throw $ErrorMsg
```

Comment:
`throw ""` will just say "Exception: " on the screen. Maybe just log the first part of Uri and not the parameters.

Reply:
This throw will be always caught and not showed to the user, and the re-throwing will be performed with a different message (the initial one could be a part of it, but an empty message is not a problem in this case). We do not want to log all these errors to the screen where they are visible, it will be confusing because some 404 errors are expected in some cases and it is perfectly fine by design.

The only goal of throwing here is to deliver the error message further, actually. I considered writing to a global variable instead, actually, but finally left this.  The decision is debatable, sure.

Reply:
Fixed. Used the `Data` property of the exception to deliver the message, as discussed.